plugins/kubernetes/src/api/KubernetesBackendClient.ts

```diff
@@ -110,4 +131,32 @@ export class KubernetesBackendClient implements KubernetesApi {
 
     return (await this.handleResponse(response)).items;
   }
+
+  async proxy(options: {
+    clusterName: string;
+    path: string;
+    init?: RequestInit;
+  }): Promise<Response> {
+    const { authProvider } = await this.getCluster(options.clusterName);
+    const k8sToken = await this.getBearerToken(authProvider);
+    const url = `${await this.discoveryApi.getBaseUrl('kubernetes')}/proxy${
+      options.path
+    }`;
+    const identityResponse = await this.identityApi.getCredentials();
+    const headers = identityResponse.token
+      ? {
+          ...options.init?.headers,
+          [`Backstage-Kubernetes-Cluster`]: options.clusterName,
+          [`Backstage-Kubernetes-Authorization`]: `Bearer ${k8sToken}`,
+          Authorization: `Bearer ${identityResponse.token}`,
+        }
+      : {
+          ...options.init?.headers,
+          [`Backstage-Kubernetes-Cluster`]: options.clusterName,
+          [`Backstage-Kubernetes-Authorization`]: `Bearer ${k8sToken}`,
+        };
+    const response = await fetch(url, { ...options.init, headers });
+
+    return this.handleResponse(response);
```

Comment:
I think think you want to use the exact same response handling here, 404 could come back for other reasons than the plugin not being installed. In general I think it might be best to just forward the response directly instead? There could be cases where you for example want to stream out data, you're not dealing with JSON, etc.

Reply:
This does make sense, i just tried to reuse some components but those edgecases definitely make me want to change from using the handleResponse function and just returning the response directly. I will rework this.

Reply:
On this same topic, I noticed the following coverage:
```console
% yarn test --watchAll=false --collectCoverageFrom '**/KubernetesBackendClient.ts' --coverage plugins/kubernetes/src/api/KubernetesBackendClient.test.ts
 PASS   @backstage/plugin-kubernetes  plugins/kubernetes/src/api/KubernetesBackendClient.test.ts
  KubernetesBackendClient
    ✓ hits the /clusters API (50 ms)
    ✓ hits the /resources/custom/query API (5 ms)
    ✓ hits the /services/{entityName} api (4 ms)
    ✓ hits the /resources/workloads/query API (3 ms)
    ✓ hits the /proxy API (5 ms)
    ✓ /proxy API throws a ERROR_NOT_FOUND if the cluster in the request is not found (5 ms)
    ✓ hits /proxy api when signed in as a guest (6 ms)

----------------------------|---------|----------|---------|---------|-------------------
File                        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------|---------|----------|---------|---------|-------------------
All files                   |   85.71 |       70 |     100 |   85.36 |
 KubernetesBackendClient.ts |   85.71 |       70 |     100 |   85.36 | 46-56
----------------------------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       7 passed, 7 total
Snapshots:   0 total
Time:        3.566 s
Ran all test suites matching /plugins\/kubernetes\/src\/api\/KubernetesBackendClient.test.ts/i
```

and it's striking that lines the uncovered lines are precisely the ones where the response from the kubernetes-backend is not successful:

https://github.com/backstage/backstage/blob/0ebbfd45c62bb05c1780a5d7184a5dae2fa0580a/plugins/kubernetes/src/api/KubernetesBackendClient.ts#L46-L56

Can we add tests to cover the cases when the relevant backend API (including proxy) returns an error status?

Reply:
yes, working on test coverage for these cases

Reply:
Done